src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -1648,14 +1688,17 @@ public void ReorderFiles(ImmutableArray<string> filePaths)
                     }
                     else
                     {
-                        _project._workspace.ApplyBatchChangeToWorkspace(solution =>
+                        // If we're passing useAsync: false, we should always get a task that was already completed.
```

Comment:
Is that true? There are other awaits in the method as well.

Reply:
Ah nm... still seems suspect though. Why don't we split into 3 methods? 

1. Sync
2. Async

3. Sync work both async and sync point to